pages/api/icons.js

```diff
@@ -0,0 +1,46 @@
+import logger from "@config/logger";
+import connectMongo from "@config/mongo";
+import Link from "@models/Link";
+
+export default async function handler(req, res) {
+  if (req.method != "GET") {
+    return res
+      .status(400)
+      .json({ error: "Invalid request: GET request required" });
+  }
+
+  const icons = await getPopularIcons();
+  return res.status(200).json(icons);
+}
+
+export async function getPopularIcons() {
+  await connectMongo();
+  let icons = [];
+  try {
+    icons = await Link.aggregate([
+      {
+        $group: {
+          _id: "$icon",
+          count: { $count: {} },
+        },
+      },
+      {
+        $sort:{
```

Comment:
I think the formatting of this code is not inline with the project, check you have Prettier VScode plugin installed, it will fix this on save